MapboxNavigation/MapboxSpeechSynthesizer.swift

```diff
@@ -0,0 +1,242 @@
+                                                 with: SpeechError.apiError(instruction: modifiedInstruction,
+                                                                            options: options,
+                                                                            underlying: urlError))
+                return
+            } else if let error = error {
+                self.delegate?.speechSynthesizer(self,
+                                                 didSpeak: modifiedInstruction,
+                                                 with: SpeechError.apiError(instruction: modifiedInstruction,
+                                                                            options: options,
+                                                                            underlying: error))
+                return
+            }
+            
+            guard let data = data else {
+                self.delegate?.speechSynthesizer(self,
+                                                 didSpeak: modifiedInstruction,
+                                                 with: SpeechError.noData(instruction: modifiedInstruction,
+                                                                          options: options))
+                return
+            }
+            
+            self.cache(data, forKey: ssmlText, with: self.locale)
+            self.safeDuckAudio(instruction: modifiedInstruction)
+            self.speak(instruction: modifiedInstruction,
+                       instructionData: data)
+        }
+        
+        audioTask?.resume()
+    }
+    
+    func speak(instruction: SpokenInstruction, instructionData: Data) {
+        
+        if let audioPlayer = audioPlayer {
+            if let previousInstrcution = previousInstruction, audioPlayer.isPlaying{
+                delegate?.speechSynthesizer(self,
+                                            didInterrupt: previousInstrcution,
+                                            with: instruction)
+            }
+            
+            deinitAudioPlayer()
+        }
+        
+        switch safeInitializeAudioPlayer(data: instructionData,
+                                         instruction: instruction) {
+        case .success(let player):
+            audioPlayer = player
+            previousInstruction = instruction
+            audioPlayer?.play()
+        case .failure(let error):
+            safeUnduckAudio(instruction: instruction)
+            delegate?.speechSynthesizer(self,
+                                        didSpeak: instruction,
+                                        with: error)
+        }
+    }
+    
+    private func downloadAndCacheSpokenInstruction(instruction: SpokenInstruction) {
+        let modifiedInstruction = delegate?.speechSynthesizer(self, willSpeak: instruction) ?? instruction
+        let ssmlText = modifiedInstruction.ssmlText
+        let options = SpeechOptions(ssml: ssmlText)
+        options.locale = locale
```

Comment:
This also needs to set the `locale` to the route’s `speechLocale`. I think this context can be plumbed through the `didPassSpokenInstructionPoint(notification:)` call.